apps/io-func-wallet-solution-backend/src/wallet-instance.ts

```diff
@@ -51,3 +62,25 @@ export const getValidWalletInstance: (
           : TE.right(walletInstance)
       )
     );
+
+export const revokeUserWalletInstancesExceptOne: (
+  userId: WalletInstance["userId"],
+  walletInstanceId: WalletInstance["id"]
+) => RTE.ReaderTaskEither<WalletInstanceEnvironment, Error, void> =
+  (userId, walletInstanceId) =>
+  ({ walletInstanceRepository }) =>
+    pipe(
+      walletInstanceRepository.getAllByUserId(userId),
+      TE.map((walletInstances) => walletInstances.map((item) => item.id)),
+      TE.map((ids) => ids.filter((id) => id !== walletInstanceId)),
```

Comment:
Why not use `RA.map` `RA.filter` ?
```suggestion
      TE.map(
        flow(
          RA.filterMap((walletInstance) =>
            walletInstance.id !== walletInstanceId ? O.some(walletInstance.id) : O.none
          )
        )
      ),
```